src/sentry/static/sentry/app/components/events/interfaces/spans/spanBar.tsx

```diff
@@ -266,18 +270,44 @@ class SpanBar extends React.Component<SpanBarProps, SpanBarState> {
     }
   };
 
+  renderSpanTreeConnector = ({hasToggler}: {hasToggler: boolean}) => {
+    const {isLast, isRoot, treeDepth, continuingTreeDepths} = this.props;
+    if (isRoot) {
+      return null;
+    }
+
+    const connectorBars: Array<JSX.Element> = [];
+    for (let step = 0; step < treeDepth; step++) {
```

Comment:
@Chrissy I can understand why you used this approach. We can simplify this to be:


```js
    const connectorBars: Array<React.ReactNode> = continuingTreeDepths.map(
      (step, index) => {
        if (step >= treeDepth) {
          return null;
        }

        const left = ((treeDepth - step) * (TOGGLE_BORDER_BOX / 2) + 1) * -1;
        return <ConnectorBar style={{left}} key={index} />;
      }
    );
```

Then `continuingTreeDepths` would need to be iterated once, rather than on every loop from 0 to `treeDepth - 1.`

Reply:
Yes! Thank you! This code got kinda microwaved over time because I switched stuff around a bunch. I was looking for a way to remove the for-loop but I think just needed a bit of outside perspective.